lib/allflex/product_mapper/allflex_product.rb

```diff
@@ -251,7 +251,7 @@ def marked?
       # the id will either be defined in the maufacturing definition,
       # or it will need to be build algorithmically from the manufacturing definition and the components
       def generate_id
-        return Failure(ordered_id_components: [{ code: 'No ordered product id components passed in', errors: [{ message: 'Error in generate_id method in allflex product' }], context: {} }]) unless ordered_id_components
+        return Failure(ordered_id_components: [{ code: :not_found, errors: { message: 'Error in generate_id method in allflex product. Missing ordered_id_components.' }, context: {} }]) unless ordered_id_components
```

Comment:
The unique key needs to be inside the `:errors` and `:context` hashes

```suggestion
        return Failure(code: :not_found, errors: { ordered_id_components: { generate_id: 'Error in generate_id method in allflex product. Missing ordered_id_components.' } }, context: { ordered_id_components: {} } }) unless ordered_id_components
```

Reply:
This key wont necessarily be unique. Eg the product_component failure can be thrown on the different components on a product if invalid.

Reply:
Eg
```
[Failure({:code=>:not_found, :errors=>{:product_component=>{:message=>"Product component not found in ProductBuilder"}}, :context=>{:product_component=>{:slug=>"allflex-fdx-replacement-eid-tag-female", :id_tree=>[{:id=>"allflex-fdx-replacement-eid-tag", :props=>[{"id"=>"urn:lic:configreq:species", "value"=>"urn:lic:configreq:species:cattle"}, {"id"=>"urn:lic:configreq:nait_production_type", "value"=>"urn:lic:configreq:nait_production_type:dairy"}, {"id"=>"urn:lic:configreq:nait_location", "value"=>"12406050"}], :colour=>nil, :size=>nil}]}}}),
 Failure({:code=>:not_found, :errors=>{:product_component=>{:message=>"Product component not found in ProductBuilder"}}, :context=>{:product_component=>{:slug=>"allflex-fdx-replacement-eid-tag-male", :id_tree=>[{:id=>"allflex-fdx-replacement-eid-tag", :props=>[{"id"=>"urn:lic:configreq:species", "value"=>"urn:lic:configreq:species:cattle"}, {"id"=>"urn:lic:configreq:nait_production_type", "value"=>"urn:lic:configreq:nait_production_type:dairy"}, {"id"=>"urn:lic:configreq:nait_location", "value"=>"12406050"}], :colour=>nil, :size=>nil}]}}})]
```

```
failures.collect {|x| x.failure[:errors] }
=> [{:product_component=>{:message=>"Product component not found in ProductBuilder"}}, 
 {:product_component=>{:message=>"Product component not found in ProductBuilder"}}]
```